lib/OpenQA/Schema/Result/ScheduledProducts.pm

```diff
@@ -220,6 +220,12 @@ sub _schedule_iso {
         };
     }
 
+    # do not pass any arguments starting with "__" to jobs so these settings can be used to store additional information
+    # about the scheduled product itself
+    for my $arg (keys %$args) {
+        delete $args->{$arg} if index($arg, '__') == 0;
+    }
+
```

Comment:
As the surrounding method is rather big already I suggest to extract that logic into a function with a good name and call it here.